packages/backend-proxy-middleware/src/base/proxy.ts

```diff
@@ -0,0 +1,347 @@
+        (system.username || process.env.FIORI_TOOLS_USER) &&
+        (system.password || process.env.FIORI_TOOLS_PASSWORD)
+    ) {
+        proxyOptions.auth = `${system.username || process.env.FIORI_TOOLS_USER}:${
+            system.password || process.env.FIORI_TOOLS_PASSWORD
+        }`;
+    }
+}
+
+/**
+ * Generate options for the proxy middleware based on the input.
+ *
+ * @param backend backend system specific configuration
+ * @param options optional base options for the http-proxy-middleware
+ * @param logger optional logger instance
+ * @returns options for the http-proxy-middleware
+ */
+export async function generateProxyMiddlewareOptions(
+    backend: BackendConfig,
+    options: Options = {},
+    logger: Logger = new ToolsLogger()
+): Promise<Options> {
+    await initI18n();
+    dotenv.config();
+    // add required options
+    const proxyOptions: Options & { headers: object } = { headers: {}, ...options };
+    proxyOptions.changeOrigin = true;
+    proxyOptions.onError = proxyOptions.onError ?? ProxyEventHandlers.onError;
+    proxyOptions.onProxyReq = proxyOptions.onProxyReq ?? ProxyEventHandlers.onProxyReq;
+    proxyOptions.onProxyRes = proxyOptions.onProxyRes ?? ProxyEventHandlers.onProxyRes;
+
+    // overwrite url if running in AppStudio
+    if (isAppStudio()) {
+        const destBackend = backend as DestinationBackendConfig;
+        destBackend.destination = destBackend.destination ?? process.env.FIORI_TOOLS_DESTINATION;
+        if (destBackend.destination) {
+            await enhanceConfigsForDestination(proxyOptions, destBackend);
+            logger.info('Using destination: ' + destBackend.destination);
+        }
+    } else {
+        const localBackend = backend as LocalBackendConfig;
+        proxyOptions.target = localBackend.url;
+        // check if system credentials are stored in the store
+        const systemStore = await getService<BackendSystem, BackendSystemKey>({ logger, entityName: 'system' });
+        const system = await systemStore.read(
+            new BackendSystemKey({ url: localBackend.url, client: localBackend.client })
+        );
+        if (system) {
+            await enhanceConfigForSystem(proxyOptions, system, backend.scp, (refreshToken?: string) => {
+                if (refreshToken) {
+                    logger.info('Updating refresh token for: ' + localBackend.url);
+                    systemStore.write({ ...system, refreshToken });
+                }
+            });
+        }
+    }
+
+    proxyOptions.pathRewrite = PathRewriters.getPathRewrite(backend, logger);
+
+    if (backend.apiHub) {
+        const apiHubKey = await getApiHubKey(logger);
+        if (apiHubKey) {
+            proxyOptions.headers['apikey'] = apiHubKey;
+        }
+    }
+
+    backend.proxy = getCorporateProxyServer(backend.proxy);
+    if (backend.proxy && !isHostExcludedFromProxy(proxyOptions.target as string)) {
+        proxyOptions.agent = new (HttpsProxyAgent as any)(backend.proxy);
```

Comment:
I believe doing `import { HttpsProxyAgent } from 'https-proxy-agent';` above, will resolve the need of doing `new (HttpsProxyAgent as any)`

Reply:
Thanks! I was wondering what I did wrong here!